scipy/stats/_rvs_sampling.py

```diff
@@ -40,132 +39,12 @@ def rvs_ratio_uniforms(pdf, umax, vmin, vmax, size=1, c=0, random_state=None):
-    lies in `A` whenever it lies in `R` as the number of required
-    iterations becomes too large otherwise. To be more precise, note that
-    the expected number of iterations to draw `(U, V)` uniformly
-    distributed on `R` such that `(U, V)` is also in `A` is given by
-    the ratio ``area(R) / area(A) = 2 * umax * (vmax - vmin) / area(pdf)``,
-    where `area(pdf)` is the integral of `pdf` (which is equal to one if the
-    probability density function is used but can take on other values if a
-    function proportional to the density is used). The equality holds since
-    the area of `A` is equal to 0.5 * area(pdf) (Theorem 7.1 in [1]_).
-    If the sampling fails to generate a single random variate after 50000
-    iterations (i.e. not a single draw is in `A`), an exception is raised.
-
-    If the bounding rectangle is not correctly specified (i.e. if it does not
-    contain `A`), the algorithm samples from a distribution different from
-    the one given by `pdf`. It is therefore recommended to perform a
-    test such as `~scipy.stats.kstest` as a check.
-
-    References
-    ----------
-    .. [1] L. Devroye, "Non-Uniform Random Variate Generation",
-       Springer-Verlag, 1986.
-
-    .. [2] W. Hoermann and J. Leydold, "Generating generalized inverse Gaussian
-       random variates", Statistics and Computing, 24(4), p. 547--557, 2014.
-
-    .. [3] A.J. Kinderman and J.F. Monahan, "Computer Generation of Random
-       Variables Using the Ratio of Uniform Deviates",
-       ACM Transactions on Mathematical Software, 3(3), p. 257--260, 1977.
-
-    Examples
-    --------
-    >>> import numpy as np
-    >>> from scipy import stats
-    >>> rng = np.random.default_rng()
-
-    Simulate normally distributed random variables. It is easy to compute the
-    bounding rectangle explicitly in that case. For simplicity, we drop the
-    normalization factor of the density.
-
-    >>> f = lambda x: np.exp(-x**2 / 2)
-    >>> v_bound = np.sqrt(f(np.sqrt(2))) * np.sqrt(2)
-    >>> umax, vmin, vmax = np.sqrt(f(0)), -v_bound, v_bound
-    >>> rvs = stats.rvs_ratio_uniforms(f, umax, vmin, vmax, size=2500,
-    ...                                random_state=rng)
-
-    The K-S test confirms that the random variates are indeed normally
-    distributed (normality is not rejected at 5% significance level):
-
-    >>> stats.kstest(rvs, 'norm')[1]
-    0.250634764150542
-
-    The exponential distribution provides another example where the bounding
-    rectangle can be determined explicitly.
-
-    >>> rvs = stats.rvs_ratio_uniforms(lambda x: np.exp(-x), umax=1,
-    ...                                vmin=0, vmax=2*np.exp(-1), size=1000,
-    ...                                random_state=rng)
-    >>> stats.kstest(rvs, 'expon')[1]
-    0.21121052054580314
-
+    Please refer to `scipy.stats.sampling.RatioUniforms` for the documentation
+    and use ``RatioUniforms`` instead of this function which is deprecated.
```

Comment:
Typically, at least in recent deprecations, we have left the documentation as is but add a `.. deprecated::` directive at the top of the docstring with a similar message to the warning

Reply:
thanks for pointing this out, I will add ``.. deprecated::``. would it be ok to refer to the documentation of the new function? the main parameters are still documented, only Notes and Examples are missing. I thought it avoids duplication and creates an incentive to use the new functionality. I was also unsure if the deprecation warning would cause problems in the doc build process when the deprecated version is used, though that might not be the case.